sys/net/application_layer/nanocoap/nanocoap.c

```diff
@@ -982,6 +982,50 @@ ssize_t coap_opt_finish(coap_pkt_t *pkt, uint16_t flags)
     return pkt->payload - (uint8_t *)pkt->hdr;
 }
 
+ssize_t coap_opt_remove(coap_pkt_t *pkt, uint16_t opt_num)
+{
+    coap_optpos_t *optpos = pkt->options;
+    coap_optpos_t *prev_opt = NULL;
+    unsigned opt_count = pkt->options_len;
+    uint8_t *pkt_start = (uint8_t *)pkt->hdr;
+    size_t removed = 0;
+
+    while (opt_count--) {
+        if ((removed == 0) && (optpos->opt_num == opt_num)) {
+            prev_opt = optpos;
+            optpos++;
+            removed = optpos->offset - prev_opt->offset;
+        }
+        if ((removed > 0) && (optpos->opt_num >= opt_num)) {
+            int option_len;
+            uint16_t delta;
+            uint8_t *data = _parse_option(pkt, pkt_start + optpos->offset, &delta, &option_len);
+
+            if (data == NULL) {
+                return -EINVAL;
+            }
+            coap_put_option(pkt_start + prev_opt->offset, prev_opt->opt_num, optpos->opt_num, data,
```

Comment:
coap_put_option invokes memcpy, which has restructs in place that are not met; the data would need to be memmove'd.

(That might be easier done in bulk than looping).

Reply:
> […] which has restructs in place that are not met;

Not sure what you mean by that.